### DISCUSSION TITLE
Help with programmatically setting options for ManagedDockWindow

### DISCUSSION THREAD

**Discussion by EliasDubroff** [pymeasure/pymeasure#1139] [Q&A]
Hi there, I have been using the ManagedDockWindow to graph a voltage reading off of my device and wish to do this continually. Because of this I have been trying to find a way to have the X-axis set to the correct range and to have the auto-pan setting turned on when I run the program. In the documentation, it says that there is a way to do so using the pyqtgraph's PlotItem API, but so far I have not been able to get it to work correctly. I have looked through the examples and did not see anything that related to how to implement this. I was wondering if anyone could help me with figuring out how to do so? Thank you so much in advance your help is greatly appreciated!

**Comment by CasperSchippers** [comment]
If I understand correctly, you want to fix the range of the x-axis, and leave the range of the y-axis at automatic?

From within the managed window, you can set the x-range of the plot with the method `self.widget_list[0].plot.setXRange` (`self.widget_list[0].plot.setXRange(0, 10)` sets the x-range from 0 to 10).

`self.widget_list[0].plot` is the [plotitem](https://pyqtgraph.readthedocs.io/en/latest/api_reference/graphicsItems/plotitem.html) (at least, for the default `ManagedWindow`, if you use one of the other managed-windows, or change the tabs, the index on widget_list will differ).

On this plotitem, you can change the x-range with the [`setXRange`](https://pyqtgraph.readthedocs.io/en/latest/api_reference/graphicsItems/viewbox.html#pyqtgraph.ViewBox.setXRange) method, the y-range with the [`setYRange`](https://pyqtgraph.readthedocs.io/en/latest/api_reference/graphicsItems/viewbox.html#pyqtgraph.ViewBox.setYRange) method, or both with the [`setRange`](https://pyqtgraph.readthedocs.io/en/latest/api_reference/graphicsItems/viewbox.html#pyqtgraph.ViewBox.setRange) method. (The docs for the latter explain the arguments for the former two.)

Hope this helps!

**Comment by EliasDubroff** [comment]
Hi this definitely helps thank you very much. I am trying to use the ManagedDockWindow instead of the ManagedWindow like you were talking about. Looking at the code of the ManagedDockWindow, there is no plot_widget within the widget_list. Is there a way to add the plot_widget to this list so that I may use it in my GUI, when I have tried to do this I have been given an error?

**Reply by CasperSchippers** [reply]
In that case, I don't think you want to add a plot-widget to the widget list, since this means you'll add another tab, instead of modifying the existing plot. The main thing you need is to access the plot-item(s) (with the setXRange ... methods) that are contained within the dock widget.

I'll have to check (can do that later today, probably in the evening) how to do that.

**Reply by EliasDubroff** [reply]
Okay thank you very much that would be super helpful.

**Reply by CasperSchippers** [reply]
The dock-widget has a list `plot_frames` which contains all the `PlotFrames` (I'm assuming you have multiple, since you're using the ManagedDockWindow?); on a `PlotFrame` you can access the pyqtgraph `plotitem` via the `plot` attribute. So, for exmple, you can use the method `self.dock_widget.plot_frames[0].plot.setXRange` to set the x-range for the 1st plot in the dock-widget.
